strongbox-web-core/src/main/java/org/carlspring/strongbox/security/vote/CustomAccessDecisionVoter.java

```diff
@@ -0,0 +1,123 @@
+package org.carlspring.strongbox.security.vote;
+
+import org.carlspring.strongbox.security.user.SpringSecurityUser;
+import org.carlspring.strongbox.users.domain.Features;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.ProxyMethodInvocation;
+import org.springframework.security.access.AccessDecisionVoter;
+import org.springframework.security.access.ConfigAttribute;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.web.FilterInvocation;
+
+/**
+ * Customization of {@link AccessDecisionVoter} according to custom access strategies for users
+ * like per-repository access etc.
+ *
+ * @author Alex Oreshkevich
+ * @see https://youtrack.carlspring.org/issue/SB-603
+ */
+public class CustomAccessDecisionVoter
+        implements AccessDecisionVoter, AuthenticationProvider
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(CustomAccessDecisionVoter.class);
+
+    private Authentication authentication;
```

Comment:
> We can't hardcode URL's or any path's or anything because they are all subject of change.

i suppose that reposytory and storage urls will not change